src/strtab.rs

```diff
@@ -60,37 +55,54 @@ impl<'a> Strtab<'a> {
                 overflow
             )));
         }
-        Ok(Strtab {
-            bytes: &bytes[offset..end],
-            delim: ctx::StrCtx::Delimiter(delim),
-        })
+        let mut result = Self::from_slice_unsafe(bytes, offset, len, delim);
+        {
+            let mut i = 0;
+            while i < len {
+                let string = get_str(i, result.bytes, result.delim)?;
```

Comment:
this is stricter now, since any invalid string will cause failure at construction, instead of at `get` time, but this is the price we pay, i suppose ! :)

Reply:
> which in turn also moves the error handling with it

As I already remarked in the opening message.